File: java/Kode/Staff.java
public class Staff extends Weapon {
    private double spellingSpeed;
    private String element;

    public Staff(String name, int level, String tier, WeaponMaterial material, 
                 double spellingSpeed, String element) {
        super(name, level, tier, material);
        this.spellingSpeed = spellingSpeed;
        this.element = element;
    }

    @Override
    protected String specificsInfo() {
        return "Element: " + element + ", Spelling Speed: " + spellingSpeed;
    }

    // Getter untuk kebutuhan Hybrid
    public double getSpellingSpeed() { return spellingSpeed; }
    public String getElement() { return element; }
}

File: cpp/Kode/Staff.cpp
#include "Staff.h"
#include "WeaponMaterial.h" 

Staff::Staff(std::string n, int l, std::string t, const std::vector<std::pair<std::string, double>>& s, 
      std::unique_ptr<WeaponMaterial> m,
      double ss, std::string e) : 
    Weapon(n, l, t, s, std::move(m)), spellingSpeed(ss), Element(e) {}

void Staff::displayWeapon() const {
    Weapon::displayWeapon(); 
    std::cout << "  Specifics: Element: " << Element << ", Spelling Speed: " << spellingSpeed << std::endl;
}

void Staff::displayWeaponWithoutMaterial() const {
    Weapon::displayWeaponWithoutMaterial(); 
    std::cout << "  Specifics: Element: " << Element << ", Spelling Speed: " << spellingSpeed << std::endl;
}